projects/ngx-avatar/src/lib/avatar.service.ts

```diff
@@ -81,6 +84,18 @@ export class AvatarService {
     return [AvatarSource.INITIALS, AvatarSource.VALUE].includes(sourceType);
   }
 
+  private getKey(source: Source): string {
```

Comment:
I disagree, because the knowledge of source unicity does not live in the source itself, but rather in classes that create and handle them. So that would be in `AvatarService`, which need a unicity by type and id, and `AvatarComponent.addSource()`, which need a unicity by type (without id).

Moving this method to the interface, would force all implementations to duplicate the same method body. Or alternatively we could transform the interface into an abstract class, but I assumed it would be too big a change. But that might be the best solution if you don't want to keep this in the service.

Reply:
Yeah, it's so implicit that I even wrote "inside the source abstract class" for the first time, then I corrected myself. I would go with renaming the method in this case, to avoid such change.

Reply:
done